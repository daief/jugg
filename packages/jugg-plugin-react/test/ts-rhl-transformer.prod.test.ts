import * as ts from 'typescript';
import { join } from 'path';
import transformerFactory from '../src/ts-rhl-transformer/ts-rhl-transformer.prod';
import { readdirSync, statSync, readFileSync } from 'fs';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');

describe('ts-rhl-transformer-prod', () => {
  describe('drop-hot', () => {
    const fixturesDir = join(FIXTURES_DIR, 'prod');

    readdirSync(fixturesDir).forEach(fixtureName => {
      const fixtureFile = join(fixturesDir, fixtureName);

      if (statSync(fixtureFile).isFile()) {
        it(fixtureName, () => {
          const sourceCode = readFileSync(fixtureFile, 'utf-8');
          const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
          const result = ts.transform(source, [transformerFactory()]);
          const transformedSourceFile = result.transformed[0];
          const resultCode = printer.printFile(transformedSourceFile);
          expect(resultCode).toMatchSnapshot();
        });
      }
    });
  });
});
