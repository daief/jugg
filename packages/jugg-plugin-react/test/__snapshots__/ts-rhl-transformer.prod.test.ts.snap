// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-rhl-transformer-prod drop-hot drop-hot.prod.js 1`] = `
"import React from 'react';
import { hot, foo } from 'react-hot-loader';
import { hot as namedHot, foo as namedFoo } from 'react-hot-loader';
import { hot as namedHot2 } from 'react-hot-loader';
import { hot as notRHLHot } from 'not-react-hot-loader';
import * as RHL from 'react-hot-loader';
import * as RHL2 from 'react-hot-loader';
import * as NOTRHL from 'not-react-hot-loader';
const App = () => <div>Hello World!</div>;
const a = App;
const b = App;
const c = App;
const d = App;
const e = App;
foo(module)(App);
notRHLHot(module)(App);
namedFoo(module)(App);
RHL.foo(module)(App);
NOTRHL.hot(module)(App);
export { a, b, c, d, e };
"
`;
